lib/rouge/lexers/eex.rb

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class EEX < TemplateLexer
+      title "EEX"
+      desc "Embedded Elixir"
+
+      tag 'eex'
+
+      filenames '*.eex'
+
+      def initialize(opts={})
+        @elixir_lexer = Elixir.new(opts)
+
+        super(opts)
+      end
+
+      start do
+        parent.reset!
+        @elixir_lexer.reset!
+      end
+
+      open  = /<%%|<%=|<%#|<%/
+      close = /%%>|%>/
+
+      state :root do
+        rule %r/<%#/, Comment, :comment
+
+        rule open, Comment::Preproc, :elixir
+
+        rule %r/.+?(?=#{open})|.+/m do
```

Comment:
This is an optimisation we need to make in the ERB lexer but this regex pattern should use the `o` option to avoid multiple interpolations.

Reply:
ok, noted